src/ConfigTool.h

```diff
@@ -2,38 +2,52 @@
 	Name:	ConfigTool.h
 	Author:	Tvde1
 */
-
-#include <Arduino.h>;
-#include <map>;
-#include "ArduinoJson.h"
-#include "ESP8266WebServer.h";
-#include <string>;
-
 #ifndef _ConfigTool_h
 #define _ConfigTool_h
 
+#include <map>
+#include <ArduinoJson.h>
+
+#if defined(ARDUINO_ARCH_ESP8266) //ESP8266
+    #include <ESP8266WebServer.h>
+	#define WebServer ESP8266WebServer
+#elif defined(ARDUINO_ARCH_ESP32) //ESP32
+    #include <SPIFFS.h>
+    #include <WebServer.h>
+#else
+	#error "unsupported architecture - intended for ESP32 and ESP8266"
+#endif
+
+#define VT_TEXT 1
+#define VT_BOOL 2
+#define VT_NUM 3
+#define VT_IP 4
+
 struct BaseVar {
 	String name;
-	virtual void serialize(JsonObject*) = 0;
-	virtual void deserialize(JsonObject*) = 0;
+	bool hideInWeb = false;
+	bool hideValueInWeb = false;
```

Comment:
What's the use cases of `hideInWeb` and `hideValueInWeb`? Why add it to the config tool if it won't be changed?

I could see a use case for showing the value but making it read-only (e.g. textbox is greyed out). That way you can show like the hostname/MAC of the ESP which could help in the process of changing the configuration.